external/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisBackedBlockRDD.scala

```diff
@@ -147,6 +152,14 @@ class KinesisSequenceRangeIterator(
   private var lastSeqNumber: String = null
   private var internalIterator: Iterator[Record] = null
 
+  // variable for kinesis wait time interval between next retry
+  private var kinesisWaitTimeMs = JavaUtils.timeStringAsMs(
```

Comment:
I don't think you want to do this-- ```kinesisWaitTimeMs``` is never reset to the default value after the retry loop exists. I think you should make this a ```val``` and introduce a ```var``` initialized to its value within ```retryOrTimeout()``` to store the wait time for each retry iteration.